python/tank/authentication/shotgun_authenticator.py

```diff
@@ -153,9 +151,11 @@ def _create_session_user(self, login, session_token=None, password=None, host=No
         http_proxy = http_proxy or self._defaults_manager.get_http_proxy()
 
         # Create a session user
-        return user.ShotgunUser(
-            user_impl.SessionUser(host, login, session_token, http_proxy, password=password, cookies=cookies, saml_expiration=saml_expiration)
-        )
+        impl = user_impl.SessionUser(host, login, session_token, http_proxy, password=password, cookies=cookies)
+        if cookies:
```

Comment:
IIRC here i think we discussed "sharpening up" the test a bit and introspecting the contents of the cookies not just check for the presence of them. I am guessing somerthing like `if cookies and "special_sso_token" in cookies:` - does that sound right? I think that may help with future robustness if we ever need to look in cookies for anything else..